src/txmempool.cpp

```diff
@@ -1124,11 +1108,10 @@ CTxMemPool::ReadFeeEstimates(CAutoFile& filein)
     try {
         int nVersionRequired, nVersionThatWrote;
         filein >> nVersionRequired >> nVersionThatWrote;
-        if (nVersionRequired > CLIENT_VERSION)
+        if (nVersionRequired > std::max(FEE_ESTIMATES_WITHOUT_PRIORITY_VERSION, CLIENT_VERSION))
```

Comment:
I agree this works, because the old format is read `if (nFileVersion < FEE_ESTIMATES_WITHOUT_PRIORITY_VERSION)`.